orchestration/templates/process-xml-archive.yaml

```diff
@@ -23,18 +24,30 @@ spec:
           {{- $gcsPrefix := "{{inputs.parameters.gcs-prefix}}" }}
       dag:
         tasks:
-          # TODO: Query the processing_history table to see if we've already
+          # Query the processing_history table to see if we've already
           # done the work to ingest the release-date under the current pipeline
           # version.
           #
           # This is trickier than it sounds because our current dev setup causes
           # the pipeline version to always be "latest", which isn't very useful.
           #
-          # A plausible approach would be to only run the pre-check if a version
-          # is pinned, though that will make wiring up the Argo dependencies more
-          # complicated.
+          # For now we assume that "latest" means we should always reprocess the
+          # data, and disable auto-snapshot in that case.
+          {{- if $versionIsPinned }}
+          - name: check-processing-history
+            template: get-processing-history-rows
+            arguments:
+              parameters:
+                - name: release-date
+                  value: {{ $releaseDate | quote }}
+          {{- end }}
+          {{- $processingNeeded := "{{tasks.check-processing-history.outputs.result}} == 0" }}
```

Comment:
This file is now pretty big and nasty. I'd like to explore more ways to break it down, but would rather merge first to pin the working functionality